lib/anyway/config.rb

```diff
@@ -126,14 +128,36 @@ def config_attributes
         end
       end
 
-      def required(*names)
-        unless (unknown_names = (names - config_attributes)).empty?
-          raise ArgumentError, "Unknown config param: #{unknown_names.join(",")}"
-        end
+      def required(*names, env: nil)
+        unknown_names = names - config_attributes
+        raise ArgumentError, "Unknown config param: #{unknown_names.join(",")}" if unknown_names.any?
 
+        names = required_with_env(names, env) if env
```

Comment:
Let's call it `#filter_by_env`

```suggestion
        names = filter_by_env(names, env) if env
```

And we can probably move `if env` into the method (since we already check for different possible values there).